### ISSUE TITLE
testing performance

### ISSUE THREAD

**Issue by InstantWindy** [naoto0804/pytorch-AdaIN#10]
Hello,Did you train according to your train.py file? The total number of iterations is 100,000.I found that you trained according to your train.py file, and the final test effect failed to reach the result of the paper.
How was your test effect? Thank!

**Comment by naoto0804** [comment]
The original implementation used 160000 iterations. Thank you for pointing out.
I agree that the longer period of training will produce better results.
https://github.com/xunhuang1995/AdaIN-style/blob/master/train.lua#L38

I accidentally deleted the checkpoint of the model so I cannot show the results, I'm sorry.
How bad was the result compared to the paper? Could you illustrate some bad results?

**Comment by InstantWindy** [comment]
Does vgg19 not train during training? I had a total of 200000 iterations, but this result is still not good.I set alpha=0.75,get the test result:
![2007_000032_stylized_flower_of_life](https://user-images.githubusercontent.com/23608180/40096439-71489b64-5903-11e8-9eb6-c6bd35ab0d26.jpg)

**Comment by InstantWindy** [comment]
The style is flower_of_life，just one style.I set the initial learing rate is 1e-4,batch_size=12,adopting lr decay:

def adjust_learning_rate(optimizer, iteration_count):
    """Imitating the original implementation"""
    lr = init_lr / (1.0 + lr_decay * iteration_count)
    for param_group in optimizer.param_groups:
        param_group['lr'] = lr


I don't know why ,could you help me ? Thanks!

**Comment by InstantWindy** [comment]
I use VOC2012 as the content image. Do you mean that my training content pictures and style pictures are very close to result in poor training? But must this training content use the COCO dataset? However, it is not clearly stated in the paper whether VGG19 is trained during training. I think your code says that vgg19 is not trained because you set requirements_grad=False.

**Comment by naoto0804** [comment]
No, I mean, the image used for your ** testing** are close to each other. If possible please test other combinations used in the original paper.

As I read the original codes, it seems that the loss is not back-propagated to **encoder**
https://github.com/xunhuang1995/AdaIN-style/blob/facb6b619d51564fd5040ba71d15c980a889dddc/train.lua#L270-L271

**Comment by InstantWindy** [comment]
Ok.Thank you very much! When you trained ,did you get the paper result?

**Comment by naoto0804** [comment]
Oh, I recall that using higher weight for style loss such as `--style_weight 10.0` produced perceptually better results, which mean transferring detailed strokes and patterns.

**Comment by naoto0804** [comment]
Finally, I recall that using higher weight for style loss such as --style_weight 10.0 produced perceptually better results.
Here I show the results using the combination shown here:
https://github.com/xunhuang1995/AdaIN-style#examples
![results](https://user-images.githubusercontent.com/7067992/40209477-5b7f32b6-5a7a-11e8-8904-c66e463cb2f7.png)

**Comment by InstantWindy** [comment]
Did you just change this style-weight? So is this style-weight value too small, leading to poor results?Thank you very much!

**Comment by naoto0804** [comment]
Yes, the smaller style-weight made the decoder focus more on the content.

**Comment by InstantWindy** [comment]
Thank you very much!
I think it is strange that the training encoder seems to have nothing to do with the content image, because I was using the weight of the encoder training you provided to test and found that the effect is similar to the paper. But the test data I use is VOC2012, and you use the MSCOCO dataset for training。

**Comment by InstantWindy** [comment]
Hello！Do you know which of the style transfer can replace the background of the content image with the background of the style image?

**Comment by naoto0804** [comment]
Yes, I did not implement.

**Comment by InstantWindy** [comment]
Hello, are you a student? I am a graduate student. What is your research direction? I am doing image segmentation. I am a beginner. I think you are very helpful.Thanks!

**Comment by naoto0804** [comment]
I'm a Ph. D student. Please see here for details! https://naoto0804.github.io/

**Comment by InstantWindy** [comment]
I think the contentFeatureBG  dimension and targetFeature dimension are  different from contentFeature dimension,but the code writes: targetFeature = targetFeature:viewAs(contentFeature),I don't know 
![tim 20180519154009](https://user-images.githubusercontent.com/23608180/40270351-6c9a2bba-5bbe-11e8-8ade-b196f95ea544.png)
